### PR TITLE
Add lots of typing to layers.base

### PR DESCRIPTION
**Pull request by dstansby**
# References and relevant issues

# Description
This adds a lot more typing to `napari.layers.base`. It's not finished, but I thought it would be easier for review if I did all the low hanging fruit first.

### CODE REVIEW

**Review by brisvag — APPROVED** [review]

File: napari/layers/utils/interactivity_utils.py [hunk 1]
```diff
@@ -51,7 +52,7 @@ def displayed_plane_from_nd_line_segment(
 
 def drag_data_to_projected_distance(
     start_position, end_position, view_direction, vector
```
Comment (psobolewskiPhD) [inline napari/layers/utils/interactivity_utils.py:54]: Should these arguments get types too, e.g `np.ndarray`?

**Review by psobolewskiPhD — APPROVED** [review]
Everything looks fine to me. I did leave a small question.

File: napari/layers/utils/interactivity_utils.py (hunk 1 above)
Comment (dstansby) [reply napari/layers/utils/interactivity_utils.py:54]: Probably, but the intent here was to type `layers.base` and bits in other files that were required for that added typing, so I'd rather leave this for now.

Comment (psobolewskiPhD) [reply napari/layers/utils/interactivity_utils.py:54]: ok, and just so I further understand: `utils.interactivity_utils.py` isn't checked by mypy so the partial typing doesn't cause any squeals, right?

File: napari/layers/base/base.py [hunk 2]
```diff
@@ -734,58 +737,58 @@ def _on_editable_changed(self) -> None:
...
+    def scale(self) -> npt.NDArray:
+        """array: Anisotropy factors to scale data into world coordinates."""
         return self._transforms['data2physical'].scale
 
     @scale.setter
-    def scale(self, scale):
+    def scale(self, scale: Optional[npt.NDArray]) -> None:
         if scale is None:
-            scale = [1] * self.ndim
+            scale = np.array([1]) * self.ndim
```
Comment (Czaki) [inline napari/layers/base/base.py]: For `ndim=3` previous version will evaluate to `[1, 1, 1]`, Current will evaluate to `np.ndarray([3])` 

```suggestion
            scale = np.array([1] * self.ndim)
```

**Review by Czaki — CHANGES_REQUESTED** [review]

**Review by Czaki — APPROVED** [review]

**Pull request merged**